cmd/tailscale/main.go

```diff
@@ -858,8 +1059,108 @@ func (a *App) processUIEvents(w *app.Window, events []UIEvent, act jni.Object, s
 		case SearchEvent:
 			state.query = strings.ToLower(e.Query)
 			a.updateState(act, state)
+		case FileSendEvent:
+			a.sendFiles(e, files)
+		}
+	}
+}
+
+func (a *App) sendFiles(e FileSendEvent, files []File) {
+	go func() {
+		var totalSize int64
+		for _, f := range files {
+			totalSize += f.Size
 		}
+		if totalSize == 0 {
+			totalSize = 1
+		}
+		var totalSent int64
+		progress := func(n int64) {
+			totalSent += n
+			e.Updates(FileSendInfo{
+				State:    FileSendTransferring,
+				Progress: float64(totalSent) / float64(totalSize),
+			})
+			a.invalidate()
+		}
+		defer a.invalidate()
+		for _, f := range files {
+			if err := a.sendFile(e.Context, e.Target, f, progress); err != nil {
+				if errors.Is(err, context.Canceled) {
+					return
+				}
+				log.Printf("file send failed: %v", err)
```

Comment:
be careful about logging filenames. we scrub filenames (and even exact sizes) from all log output in the clients, so we can't know what's being shared.

Reply:
Removed logging. The UI will show the error anyway.